tests/functional/metadata/mongodb/putObject.spec.js

```diff
@@ -354,6 +357,36 @@ describe('MongoClientInterface:metadata.putObjectMD', () => {
                 ], done);
             });
 
+            it(`should fail on versionID conflict when putting a new version ${variation.it}`, done => {
+                const objVal = {
+                    key: OBJECT_NAME,
+                };
+                const params = {
+                    versioning: true,
+                    versionId: null,
+                    repairMaster: null,
+                };
+
+                // simulate a versionId collision by always generating the same versionId
+                const genVID = sinon.stub(VersionID, 'generateUniqueVersionId')
+                    .returns('test-version-id');
+
+                async.series([
+                    // We first create a master and a version
+                    next => metadata.putObjectMD(BUCKET_NAME, OBJECT_NAME, objVal, params, logger, next),
+                    // We put another version of the object with the same versionId
+                    next => metadata.putObjectMD(BUCKET_NAME, OBJECT_NAME, objVal, params, logger, next),
+                ], err => {
+                    assert(err.is.InternalError,
+                        `expected InternalError, got ${err.name} with message: ${err.message}`,
+                    );
+                    // make sure the retry triggered on the first collision detection
+                    assert(genVID.calledThrice,
+                        `expected generateUniqueVersionId to be called thrice, got ${genVID.callCount} times`);
+                    done();
```

Comment:
Can we add a variant, where we have a return being a new vID on the retry, so that we confirm, in that case, that no InternalError is returned, and the new vID is properly created during the retry?